samples/mpsl/timeslot/src/main.c

```diff
@@ -0,0 +1,229 @@
+		if (request_in_cb) {
+			/* Request new timeslot when callback returns */
+			signal_callback_return_param.params.request.p_next = &timeslot_request_normal;
+			signal_callback_return_param.callback_action = MPSL_TIMESLOT_SIGNAL_ACTION_REQUEST;
+		} else {
+			/* No callback action when callback returns, so timeslot will be ended */
+			signal_callback_return_param.params.request.p_next = NULL;
+			signal_callback_return_param.callback_action = MPSL_TIMESLOT_SIGNAL_ACTION_NONE;
+		}
+
+		p_ret_val = &signal_callback_return_param;
+		break;
+	case MPSL_TIMESLOT_SIGNAL_CANCELLED:
+		break;
+	case MPSL_TIMESLOT_SIGNAL_BLOCKED:
+		break;
+	case MPSL_TIMESLOT_SIGNAL_SESSION_CLOSED:
+		break;
+	case MPSL_TIMESLOT_SIGNAL_SESSION_IDLE:
+		break;
+	case MPSL_TIMESLOT_SIGNAL_INVALID_RETURN:
+		break;
+	default:
+		error();
+		break;
+	}
+
+	struct cb_fifo_data_t cb_data = { .cb_signal_type = signal_type };
+
+	/* Allocate memory for cb_data on heap, and put it in the FIFO. */
+	size_t size = sizeof(struct cb_fifo_data_t);
+	char *mem_ptr = k_malloc(size);
+	__ASSERT_NO_MSG(mem_ptr != 0);
+	memcpy(mem_ptr, &cb_data, size);
+	k_fifo_put(&cb_fifo, mem_ptr);
+
+	return p_ret_val;
+}
+
+static void mpsl_timeslot_demo(void)
+{
+	int err;
+	char input_char;
+
+	printk("-----------------------------------------------------\n");
+	printk("Press a key to open session and request timeslots.\n");
+	printk("* Press 'a' for a session where each request makes a new request\n");
+	printk("* Press 'b' for a session with single requests\n");
+	input_char = console_getchar();
+	printk("%c\n", input_char);
+	err = mpsl_timeslot_session_open(mpsl_timeslot_callback);
+	if (err) {
+		error();
+	}
+
+	if (input_char == 'a') {
+		request_in_cb = true;
+	} else if (input_char == 'b') {
+		request_in_cb = false;
+	}
+
+	err = mpsl_timeslot_request(&timeslot_request_earliest);
+	if (err) {
+		error();
+	}
+
+	printk("Press any key to close the session.\n");
+	console_getchar();
+
+	err = mpsl_timeslot_session_close();
+	if (err) {
+		error();
+	}
+}
+
+static void timeslot_thread(void)
+{
+	while (1) {
+		mpsl_timeslot_demo();
+		k_sleep(1000);
+	}
+}
+
+static void console_print_thread(void)
+{
+	while (1) {
+		struct cb_fifo_data_t *p_cb_data = k_fifo_get(&cb_fifo, K_FOREVER);
+		switch (p_cb_data->cb_signal_type) {
+		case MPSL_TIMESLOT_SIGNAL_START:
+			printk("Callback: Timeslot start\n");
+			break;
+		case MPSL_TIMESLOT_SIGNAL_SESSION_IDLE:
+			printk("Callback: Session idle\n");
+			break;
+		case MPSL_TIMESLOT_SIGNAL_SESSION_CLOSED:
+			printk("Callback: Session closed\n");
+			break;
+		default:
+			printk("Callback: Other signal: %d\n", p_cb_data->cb_signal_type);
+			break;
+		}
+		k_free(p_cb_data);
+	}
+}
+
+static void led_blink_thread(void)
+{
+	int blink_counter = 0;
+
+	while (1) {
+		dk_set_led(RUN_STATUS_LED, (++blink_counter) % 2);
+		k_sleep(RUN_STATUS_BLINK_INTERVAL_US);
+	}
+}
+
+static void irq_connect(void)
+{
+	IRQ_CONNECT(TIMER0_IRQn, MPSL_IRQ_LOW_PRIO,
```

Comment:
This should be priority high I think

Reply:
And maybe we should use IRQ_DIRECT_CONNECT instead?